### PR TITLE
Remove DfciPkg from MsBootPolicy

### PR DESCRIPTION
**Pull request by mikeytdisco**
# Preface

Please ensure you have read the [contribution docs](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md) prior
to submitting the pull request. In particular,
[pull request guidelines](https://github.com/microsoft/mu/blob/master/CONTRIBUTING.md#pull-request-best-practices).

## Description

Part of the plan to remove dependencies on DfciPkg from PcBdsPkg,

For each item, place an "x" in between `[` and `]` if true. Example: `[x]`.
_(you can also check items in the GitHub UI)_

- [ ] Impacts functionality?
- [ ] Impacts security?
- [x] Breaking change?
- [ ] Includes tests?
- [ ] Includes documentation?

## How This Was Tested

Tested with a build of QemuQ35Pkg.

## Integration Instructions

If your platform uses PcdBdsPkg\MsBootPolicy, change to use OemPkg/MsBootPolicy

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

**Comment by makubacki** [comment]
I'm checking breaking change for most of these since it breaks consumers that were using the code at the previous location.

**Comment by makubacki** [comment]
I updated the title from "Remove DfciPkg from MsBootPolicy" to "Remove DfciPkg references from MsBootPolicy". I think that describes the change a little more clearly to someone newly approaching it.

File: PcBdsPkg/Library/BootManagerSettingsDxeLibNull/BootManagerSettingsDxeLibNull.inf [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+# BootManagerSettingsLib - PcBds NULL BootManagerSettingsLib.
+#
+# Copyright (C) Microsoft Corporation. All rights reserved.
+# SPDX-License-Identifier: BSD-2-Clause-Patent
+##
+
+[Defines]
+  EDK_RELEASE_VERSION             = 0x00020000
+  INF_VERSION                     = 1.27
+  BASE_NAME                       = DeviceBootManagerLibNull
```
Comment (makubacki) [inline PcBdsPkg/Library/BootManagerSettingsDxeLibNull/BootManagerSettingsDxeLibNull.inf]: Shouldn't `BASE_NAME` be `BootManagerSettingsDxeLibNull`?

File: PcBdsPkg/Include/Settings/BootMenuSettings.h [hunk 2]
```diff
@@ -9,11 +9,30 @@ SPDX-License-Identifier: BSD-2-Clause-Patent
...
+WARNING:
+
+When used with DFCI, this header file MUST be included after:
+
+  #include <DfciSystemSettingTypes.h>
+
+or any other include file that includes DfciSystemSettingTypes.h.
+
+Failure to do so will cause a compile error in DfciSystemSettingTypes.h when it
+defines DFCI_SETTINT_ID_STRING.
```
Comment (makubacki) [inline PcBdsPkg/Include/Settings/BootMenuSettings.h]:
```suggestion
Failure to do so will cause a compile error in DfciSystemSettingTypes.h when it
defines DFCI_SETTING_ID_STRING.

```

**Review by makubacki — APPROVED** [review]

**Review by kuqin12 — APPROVED** [review]

File: PcBdsPkg/Library/BootManagerSettingsDxeLibNull/BootManagerSettingsDxeLibNull.inf (hunk 1 above)
Comment (mikeytdisco) [reply PcBdsPkg/Library/BootManagerSettingsDxeLibNull/BootManagerSettingsDxeLibNull.inf]: Yes, it should.  Fixed.

**Pull request merged**